trin-cli/src/main.rs

```diff
@@ -0,0 +1,137 @@
+use std::os::unix::net::UnixStream;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use thiserror::Error;
+
+use trin_core::cli::DEFAULT_WEB3_IPC_PATH;
+
+#[derive(StructOpt, Debug)]
+#[structopt(
+    name = "trin-cli",
+    version = "0.0.1",
+    author = "Ethereum Foundation",
+    about = "Run JSON-RPC commands against trin nodes"
+)]
+struct Config {
+    #[structopt(
+        default_value(DEFAULT_WEB3_IPC_PATH),
+        long,
+        help = "path to JSON-RPC endpoint"
+    )]
+    ipc: PathBuf,
+
+    #[structopt(help = "e.g. discv5_routingTableInfo", required = true)]
+    endpoint: String,
+}
+
+fn main() {
+    let Config { ipc, endpoint } = Config::from_args();
+
+    eprintln!(
+        "Attempting RPC. endpoint={} file={}",
+        endpoint,
+        ipc.to_string_lossy()
+    );
+    let mut client = match TrinClient::from_ipc(&ipc) {
+        Ok(client) => client,
+        Err(err) => {
+            eprintln!("Could not connect. err={:?}", err);
+            return;
+        }
+    };
+
+    let req = client.build_request(endpoint.as_str());
+    let resp = client.make_request(req);
+
+    match resp {
+        Err(error) => {
+            eprintln!("error: {}", error);
+        }
+        Ok(value) => {
+            // unwrap: this value is safe to serialize, it was just deserialized!
+            println!("{}", serde_json::to_string_pretty(&value).unwrap());
+        }
+    }
```

Comment:
If you like it, it looks like this could shrink a bit by [returning a result from main](https://doc.rust-lang.org/rust-by-example/error/result.html#using-result-in-main) (with a bit of error tweaking):

```suggestion
    let mut client = TrinClient::from_ipc(&ipc)?;

    let req = client.build_request(endpoint.as_str());
    let resp = client.make_request(req);

    let value = resp?;
    // unwrap: this value is safe to serialize, it was just deserialized!
    println!("{}", serde_json::to_string_pretty(&value).unwrap());
```

Or, in this kind of one-shot CLI tool, it doesn't bother me to `.unwrap()` either (as is already used elsewhere in the module).

Reply:
Great point, just pushed a much shorter version